### PR TITLE
Allow validators to delegate part of their epoch payment

### LINKED ISSUE #8733: Investigate validator/voter rewards delegation
### Expected Behavior

Direct rewards allocated to eg impact market which is tax exempt

### Current Behavior

Validators manually give 10% to charitable causes (contingency for celo foundation grants)
Classified as taxable income

### PR DESCRIPTION
**Pull request by m-chrzan**
### Description

This change allows validators to specify an address that will receive a fraction of each of their epoch payments, allowing e.g. regular donations of part of their payments.

### Other changes

Added the `to != address(0)` check to MockStableToken's `mint` function to ensure the real StableToken's behavior is correctly replicated.

### Tested

Unit tests.

### Related issues

- Fixes #8733

### Backwards compatibility

New functions and storage variables added. No backwards compatibility breaking changes.

### CODE REVIEW

File: packages/protocol/contracts/common/Accounts.sol [hunk 1]
```diff
@@ -303,6 +316,31 @@ contract Accounts is
...
+   * @param beneficiary The address that should receive a portion of vaidator
+   * payments.
+   * @param fraction The fraction of the validator's payment that should be
+   * diverted to `beneficiary` every epoch, given as FixidyLib value. Must not
+   * be greater than 1.
+   */
+  function setPaymentDelegation(address beneficiary, uint256 fraction) public {
+    require(isAccount(msg.sender), "Not an account");
+    FixidityLib.Fraction memory f = FixidityLib.wrap(fraction);
+    require(f.lt(FixidityLib.fixed1()), "Fraction must not be greater than 1");
```
Comment (yorhodes) [inline packages/protocol/contracts/common/Accounts.sol]: any reason not to allow all rewards to be delegated ie `f == 1`?

File: packages/protocol/contracts/governance/Validators.sol [hunk 2]
```diff
@@ -515,10 +515,18 @@ contract Validators is
...
+      );
+      (address beneficiary, uint256 fraction) = getAccounts().getPaymentDelegation(account);
+      uint256 delegatedPayment = remainingPayment.multiply(FixidityLib.wrap(fraction)).fromFixed();
+      uint256 validatorPayment = remainingPayment.fromFixed().sub(delegatedPayment);
       IStableToken stableToken = getStableToken();
       require(stableToken.mint(group, groupPayment), "mint failed to validator group");
       require(stableToken.mint(account, validatorPayment), "mint failed to validator account");
+      if (fraction != 0) {
+        require(stableToken.mint(beneficiary, delegatedPayment), "mint failed to delegatee");
+      }
```
Comment (yorhodes) [inline packages/protocol/contracts/governance/Validators.sol:529]: if `fraction == 1` we can skip the validator payment stuff

**Review by yorhodes — APPROVED** [review]
looks great, 1 question about full rewards delegation

File: packages/protocol/contracts/governance/Validators.sol (hunk 2 above)
Comment (m-chrzan) [reply packages/protocol/contracts/governance/Validators.sol:529]: Felt this additional check is unnecessary: `mint`ing 0 is ok and the function short circuits. The check is needed for the delegated payment to not attempt a `mint` to `address(0)`.

**Pull request merged**